donate/payments/views.py

```diff
@@ -199,41 +196,145 @@ def form_valid(self, form):
         return super().form_valid(form)
 
     def get_success_url(self):
-        return reverse('payments:card_single')
+        return reverse(f'payments:{self.payment_method}_{self.payment_frequency}')
 
 
-class SingleCardPaymentView(BraintreePaymentView):
-    template_name = 'payment/card_single_payment.html'
+class SinglePaymentView(BraintreePaymentMixin, FormView):
+    method = None
 
-    def dispatch(self, *args, **kwargs):
-        self.personal_details = self.request.session.get('personal_details')
-        # If we don't have personal_details in the session, the user
-        # hasn't completed the previous step, so send them to it.
-        if not self.personal_details:
-            return HttpResponseRedirect(reverse('payments:personal_details'))
-        return super().dispatch(*args, **kwargs)
+    def get_template_names(self):
```

Comment:
Thanks. I'm going to leave as is for now, because we are splitting the flows in the next piece of work (paypal flow will start from a different place to card flow), so this bit of code is going to change again.